### ISSUE TITLE
CLIENT_VERSION 2 single-tag HTML() problem

### ISSUE THREAD

**Issue by markusgritsch** [lona-web-org/lona#381]
Running the following script leads to an empty `<div id="lona"></div>`:
```Python
from lona.html import HTML
from lona import App, View

app = App(__file__)
app.settings.CLIENT_VERSION = 2

@app.route('/')
class HelloWorldView(View):
    def handle_request(self, request):
        return HTML(
            '<b>text</b>'
        )

app.run()
```

When using `CLIENT_VERSION = 1` it works fine. It also works when using a second tag like `HTML('<br><b>text</b>')` or using two arguments `HTML('text', '<b>text</b>')`. Using no tag at all also works: `HTML('text')`.

**Comment by fscherf** [comment]
@markusgritsch: Thanks for reporting. When i test you example, on the current master, i get this exception:
```
Traceback (most recent call last):
    File "/home/fscherf/devel/lona/lona/view_runtime.py", line 349, in start
      self.handle_response(response=response)
    File "/home/fscherf/devel/lona/lona/view_runtime.py", line 634, in handle_response
      title, data_type, data = self.document.apply(
    File "/home/fscherf/devel/lona/lona/html/document.py", line 90, in apply
      self.html._set_document(self)
    File "/home/fscherf/devel/lona/lona/html/abstract_node.py", line 107, in _set_document
      raise RuntimeError('node is no root node')
  RuntimeError: node is no root node
```
Do you see the same?

**Comment by markusgritsch** [comment]
Yes, now that I'm look at it, I get the same traceback *blush*. I was just looking at the browser-tab.

**Comment by fscherf** [comment]
@markusgritsch: I found the problem and added https://github.com/lona-web-org/lona/pull/380/commits/8fc947f0f84ed37dba76bb72bd67ce97f161faec to #380 since the PR is right on topic and still work in progress

You really found an edge-case here that my testing missed :)
The bug is only reproducible when only one HTML string is parsed, and this string contains only one HTML node.
How does this happen? The parsing code creates one, empty root node, then parses all HTML-strings, adding them as child nodes, and then returns a reference to the child-nodes list of the root node. All child-nodes have the empty root still set as parent node. When more than one node is returned, `lona.html.HTML` loops over the list of nodes and calls `Node.append()`, which calls `NodeList._prepare_node()` internally, which unmounts all nodes from the previous created root node. This does not happen when only one parsed, because `nodes[0]` just gets returned.
This means when only one node is parsed, you get a node back that itself is no root node, which triggers the lona safety mechanism that causes this exception.

The fix is pretty simple: I updated the parsing mechanism to only return already unmounted nodes. A test will follow.
